mlrun/run.py

```diff
@@ -1218,3 +1226,614 @@ def wait_for_runs_completion(runs: list, sleep=3, timeout=0, silent=False):
+    LOGGER_MAP = {
+        ArtifactType.DATASET.value: _Logger.log_dataset,
+        ArtifactType.DIRECTORY.value: _Logger.log_directory,
+        ArtifactType.FILE.value: _Logger.log_file,
+        ArtifactType.OBJECT.value: _Logger.log_object,
+        ArtifactType.PLOT.value: _Logger.log_plot,
+        ArtifactType.RESULT.value: _Logger.log_result,
+    }
+
+    class _Parser:
+        """
+        Inner static class to hold all the parsing functions - functions for parsing MLRun DataItem to the user desired
+        type.
+        """
+
+        @staticmethod
+        def parse_pandas_dataframe(data_item: DataItem) -> pd.DataFrame:
+            """
+            Parse an MLRun `DataItem` to a `pandas.DataFrame`.
+
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as a `pandas.DataFrame`.
+            """
+            return data_item.as_df()
+
+        @staticmethod
+        def parse_numpy_array(data_item: DataItem) -> np.ndarray:
+            """
+            Parse an MLRun `DataItem` to a `numpy.ndarray`.
+
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as a `numpy.ndarray`.
+            """
+            return data_item.as_df().to_numpy()
+
+        @staticmethod
+        def parse_dict(data_item: DataItem) -> dict:
+            """
+            Parse an MLRun `DataItem` to a `dict`.
+
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as a `dict`.
+            """
+            return data_item.as_df().to_dict()
+
+        @staticmethod
+        def parse_list(data_item: DataItem) -> list:
+            """
+            Parse an MLRun `DataItem` to a `list`.
+
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as a `list`.
+            """
+            return data_item.as_df().to_numpy().tolist()
+
+        @staticmethod
+        def parse_object(data_item: DataItem) -> object:
+            """
+            Parse an MLRun `DataItem` to its unpickled object. The pickle file will be downloaded to a local temp
+            directory and then loaded.
+
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as the original object that was pickled once it was logged.
+            """
+            object_file = data_item.local()
+            with open(object_file, "rb") as pickle_file:
+                obj = cloudpickle.load(pickle_file)
+            return obj
+
+    # The map to use for parsing an object by its type:
+    PARSER_MAP = {
+        str(pd.DataFrame): _Parser.parse_pandas_dataframe,
+        str(np.ndarray): _Parser.parse_numpy_array,
+        str(dict): _Parser.parse_dict,
+        str(list): _Parser.parse_list,
+        str(object): _Parser.parse_object,
+    }
+
+    def __init__(self):
+        """
+        Initialize a context handler.
+        """
+        self._context = None  # type: MLClientCtx
```

Comment:
```suggestion
        self._context: MLClientCtx = None 
```

Reply:
Old habit I have from supporting both python 2 and 3. Changed as well!